x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -632,6 +653,167 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
+          return acc.set(`${type}:${id}`, {
+            type,
+            id,
+            spaces: redactedSpaces,
+            ...(redactedSpacesWithMatchingAliases && {
+              spacesWithMatchingAliases: redactedSpacesWithMatchingAliases,
+            }),
+            inboundReferences,
+          });
+        }
+        return acc;
+      },
+      new Map<string, SavedObjectReferenceWithContext>()
+    );
+
+    return {
+      objects: Array.from(filteredAndRedactedObjectsMap.values()),
+    };
+  }
+
+  public async updateObjectsSpaces(
+    objects: SavedObjectsUpdateObjectsSpacesObject[],
+    spacesToAdd: string[],
+    spacesToRemove: string[],
+    options: SavedObjectsUpdateObjectsSpacesOptions = {}
+  ) {
+    const { namespace } = options;
+    const currentSpaceId = SavedObjectsUtils.namespaceIdToString(namespace); // We need this whether the Spaces plugin is enabled or not.
+
+    const allSpacesSet = new Set<string>([currentSpaceId, ...spacesToAdd, ...spacesToRemove]);
+    const bulkGetResponse = await this.baseClient.bulkGet(objects, { namespace });
+    const objectsToUpdate = objects.map(({ type, id }, i) => {
+      const { namespaces: spaces = [], version } = bulkGetResponse.saved_objects[i];
+      // If 'namespaces' is undefined, the object was not found (or it is namespace-agnostic).
+      // Either way, we will pass in an empty 'spaces' array to the base client, which will cause it to skip this object.
+      for (const space of spaces) {
+        allSpacesSet.add(space);
+      }
+      return { type, id, spaces, version };
+    });
+
+    const uniqueTypes = this.getUniqueObjectTypes(objects);
+    const { typeActionMap } = await this.ensureAuthorizedV2<'bulk_get' | 'share_to_space'>(
+      uniqueTypes,
+      ['bulk_get', 'share_to_space'],
+      Array.from(allSpacesSet),
+      { requireFullAuthorization: false }
+    );
+
+    // The user must be authorized to share every requested object in each of: the current space, spacesToAdd, and spacesToRemove.
+    try {
+      const spaces = this.getUniqueSpaces(currentSpaceId, ...spacesToAdd, ...spacesToRemove);
+      this.ensureAuthorizedInAllSpaces(objects, 'share_to_space', typeActionMap, spaces);
+    } catch (error) {
+      objects.forEach(({ type, id }) =>
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: SavedObjectAction.UPDATE_OBJECTS_SPACES,
+            savedObject: { type, id },
+            ...(spacesToAdd.length && { addToSpaces: spacesToAdd }),
+            ...(spacesToRemove.length && { deleteFromSpaces: spacesToRemove }),
+            error,
+          })
+        )
+      );
+      throw error;
+    }
+
+    const response = await this.baseClient.updateObjectsSpaces(
+      objectsToUpdate,
+      spacesToAdd,
+      spacesToRemove,
+      { namespace } // Intentionally omit `includeReferences`, as we've already resolved references above.
```

Comment:
I think this is a stale comment - `includeReferences` isn't a valid option anymore